pkg/pipelineascode/template.go

```diff
@@ -2,177 +2,25 @@ package pipelineascode
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
-	"strings"
 
-	"github.com/google/cel-go/cel"
-	"github.com/google/cel-go/checker/decls"
-	"github.com/google/cel-go/common/types"
-	"github.com/google/cel-go/common/types/ref"
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1"
-	"github.com/openshift-pipelines/pipelines-as-code/pkg/formatting"
-	sectypes "github.com/openshift-pipelines/pipelines-as-code/pkg/secrets/types"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/customparams"
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/templates"
 	"go.uber.org/zap"
 )
 
-func celEvaluate(expr string, env *cel.Env, data map[string]interface{}) (ref.Val, error) {
-	parsed, issues := env.Parse(expr)
-	if issues != nil && issues.Err() != nil {
-		return nil, fmt.Errorf("failed to parse expression %#v: %w", expr, issues.Err())
-	}
-
-	checked, issues := env.Check(parsed)
-	if issues != nil && issues.Err() != nil {
-		return nil, fmt.Errorf("expression %#v check failed: %w", expr, issues.Err())
-	}
-
-	prg, err := env.Program(checked, cel.EvalOptions(cel.OptOptimize))
-	if err != nil {
-		return nil, fmt.Errorf("expression %#v failed to create a Program: %w", expr, err)
-	}
-
-	out, _, err := prg.Eval(data)
-	if err != nil {
-		return nil, fmt.Errorf("expression %#v failed to evaluate: %w", expr, err)
-	}
-
-	return out, nil
-}
-
-// celFilter filters a query with cel, it sets two variables for the user to use
-// in the cel filter. body and pac.
-// body is the payload of the event, since it's already casted we un-marshall/and
-// marshall it as a map[string]interface{}
-func celFilter(query string, body any, pacParams map[string]string) (bool, error) {
-	nbody, err := json.Marshal(body)
-	if err != nil {
-		return false, err
-	}
-	var jsonMap map[string]interface{}
-	err = json.Unmarshal(nbody, &jsonMap)
-	if err != nil {
-		return false, err
-	}
-
-	mapStrDyn := decls.NewMapType(decls.String, decls.Dyn)
-	celDec, _ := cel.NewEnv(
-		cel.Declarations(
-			decls.NewVar("body", mapStrDyn),
-			decls.NewVar("pac", mapStrDyn),
-		))
-	val, err := celEvaluate(query, celDec, map[string]any{
-		"body": jsonMap,
-		"pac":  pacParams,
-	})
-	if err != nil {
-		return false, err
-	}
-
-	//nolint: gocritic
-	switch val.(type) {
-	case types.Bool:
-		if val.Value() == true {
-			return true, nil
-		}
-	}
-	return false, nil
-}
-
-// processParams will process the parameters as set in the repo.Spec CR.
-// value can come from a string or from a secretKeyRef or from a string value
-// if both is set we pick the value and issue a warning in the user namespace
-// we let the user specify a cel filter. If false then we skip the parameters.
-// if multiple params name has a filter we pick up the first one that has
-// matched true.
-func (p *PacRun) processParams(ctx context.Context, repo *v1alpha1.Repository, maptemplate map[string]string) error {
```

Comment:
I see code has been refactored and moved to other place but i see some duplicate tests in template_test.go can we remove those ?

Reply:
it test things a bit differently and having them in this place make it more robust, since pkg/pipelinesascode is central piece to the  controlller. We have another file in this package on github which test the full flow using github since we are sure on unit test all the pieces come together...